worlds/yoshisisland/level_logic.py

```diff
@@ -11,46 +15,44 @@ class YoshiLogic:
     consumable_logic: bool
     luigi_pieces: int
 
-
-    def __init__(self, world: Optional["YoshisIslandWorld"]) -> None:
+    def __init__(self, world: "YoshisIslandWorld") -> None:
         self.player = world.player
         self.boss_order = world.boss_order
-        self.luigi_pieces = world.luigi_pieces
+        self.luigi_pieces = world.options.luigi_pieces_required.value
 
-        if world:
-            if world.options.stage_logic == 0:
-                self.game_logic = "Easy"
-            elif world.options.stage_logic == 1:
-                self.game_logic = "Normal"
-            else:
-                self.game_logic = "Hard"
+        if world.options.stage_logic == StageLogic.option_strict:
+            self.game_logic = "Easy"
+        elif world.options.stage_logic == StageLogic.option_loose:
+            self.game_logic = "Normal"
+        else:
+            self.game_logic = "Hard"
 
-            self.midring_start = world.options.shuffle_midrings == 0
-            self.consumable_logic = world.options.item_logic != 0
+        self.midring_start = not world.options.shuffle_midrings
+        self.consumable_logic = not world.options.item_logic
 
-            if world.options.hidden_object_visibility >= 2:
-                self.clouds_always_visible = True
-            else:
-                self.clouds_always_visible = False
+        if world.options.hidden_object_visibility >= ObjectVis.option_clouds_only:

```

Comment:
Could this just be `self.clouds_always_visible = world.options.hidden_object_visibility >= ObjectVis.option_clouds_only`?

Reply:
Yep